test/extended/images/sample_repos.go

```diff
@@ -44,6 +44,14 @@ func NewSampleRepoTest(c SampleRepoConfig) func() {
 				err := oc.Run("new-app").Args("-f", c.templateURL).Execute()
 				o.Expect(err).NotTo(o.HaveOccurred())
 
+				// sometimes the deployment controller takes a while to start the DB deployment which
+				// results in the app coming up before the DB is available, which makes the app tests fail
+				// because they require the DB is available when the app started, not when the http request
+				// is made (they init the DB when they start, not on each request).  So here we force
+				// an immediate deployment of the DB.  It may error because there is already a deployment
+				// in progress, so do not check for an error.
+				oc.Run("deploy").Args(c.dbDeploymentConfigName, "--latest").Execute()
```

Comment:
If this becomes a long term pattern, I believe it applies to the s2i_*.go tests as well.  They leverage lang/db templates as well.  I *may* remember similar timing issues for those as well at some point (though the waters may have been muddied by other issues we have since resolved).

That said, they have not seemed to have been failing intermittently lately though.  If there is a precise explanation for why the s2i_*.go tests would not fail but the nodejs/mongo in sample_repos  does (the app builds are slower for those ??), I'm missing it.